bots-deployment-detector/src/agent.spec.ts

```diff
@@ -0,0 +1,159 @@
+import { provideHandleTransaction } from './agent';
+
+import { Finding, FindingSeverity, FindingType, HandleTransaction, ethers } from 'forta-agent';
+
+import { createAddress } from 'forta-agent-tools';
+import { TestTransactionEvent } from 'forta-agent-tools/lib/test';
+
+import { eventAgentEnabled, eventAgentUpdated, createAgentABI } from './utils';
+
+describe('Bot Deployment Detector Agent', () => {
+  let handleTransaction: HandleTransaction;
+
+  const mockDeployerAdddress = createAddress('0x4');
+  const mockContractAddress = createAddress('0x5');
+  const mockRandomAddress = createAddress('0x8');
+
+  const mockAgentId = '34567';
+  const mockChainId = 137;
+  const eventTypes = ['AgentCreated', 'AgentUpdated', 'AgentEnabled', 'AgentDisabled'];
+
+  beforeAll(() => {
+    handleTransaction = provideHandleTransaction(mockDeployerAdddress, mockContractAddress);
+  });
+
+  describe('handleTransaction', () => {
+    describe('Ignores Transaction', () => {
+      it('ignores any transaction that is not a from deployer Address', async () => {
+        const txEvent = new TestTransactionEvent();
+        const findings = await handleTransaction(txEvent);
+
+        expect(findings).toStrictEqual([]);
+      });
+
+      it('ignores a transaction that is from a deployer Address but not a bot deployment', async () => {
+        const txEvent = new TestTransactionEvent();
+        txEvent.setFrom(mockDeployerAdddress).setTo(mockRandomAddress).setValue('100000');
+
+        const findings = await handleTransaction(txEvent);
+
+        expect(findings).toStrictEqual([]);
+      });
+
+      it('ignores a transaction that is from a deployer Address and bot deployment but to a different bot registry/contract', async () => {
```

Comment:
in the definition you say "bot deployment" but you add agent update event. there's a confusion. in order to test bot deployment (create a bot) you need to use `addTraces` method of `TestTransactionEvent`